python/pylibcudf/pylibcudf/libcudf/utilities/stream_pool.pxd

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+from pylibcudf.libcudf.utilities.span cimport host_span
+from rmm.librmm.cuda_stream_view cimport cuda_stream_view
+
+
+cdef extern from "cudf/detail/utilities/stream_pool.hpp" namespace "cudf::detail" nogil:
```

Comment:
Not sure if this was discussed elsewhere, but pylibcudf doesn't really expose detail APIs since those can be considered private. But can we guarantee stability with `join_streams`?

Reply:
I was wondering about this as well :)

@vyasr suggested that this might be OK, but I don't think I realized at the time that the cudf API was in the `detail` namespace.

Reply:
RMM already [exposes a stream pool to Cython](https://github.com/rapidsai/rmm/blob/c6004e973e82fe00cf64ed7edc4c0884d0333ae5/python/rmm/rmm/librmm/cuda_stream_pool.pxd#L18) so the main question here is whether the consuming application needs to be using _libcudf's internal stream pool_, or just "some" stream pool that it can construct and manage itself.

Reply:
@bdice we do need this for the cudf-polars / rapidsmpf work, so if there's anything I can do to alleviate your concerns (make the pylibcudf implementation private? put it in a namespace other than `stream_pool`?) then let me know.

Reply:
Based on some offline discussion, we opted for `plc.experimental.join_streams`. https://github.com/rapidsai/cudf/pull/20316/commits/be881c50b1d36fb6e7f1b2cdce7eb6e0c7d2b5c4 implemented that.